src/color/util.ts

```diff
@@ -74,7 +74,20 @@ export const fromAlphaToFloat = (value = '1') =>
   parseFloat(value) / (value.endsWith('%') ? 100 : 1);
 
 /**
- * Convert a value ∈ [0, 255] to hex
+ * Convert a value in the inclusive range [0, 255] to hex
  */
 export const hexify = (value: number) =>
   Math.min(value, 255).toString(16).toUpperCase().padStart(2, '0');
+
+/**
+ * Calculate the grey average value for rgb and pass through alpha
+ */
+export const greyAverage = (
+  r: number,
+  g: number,
+  b: number,
+  a = 1
+): [average: number, alpha: number] => [
+  Math.round(r * 0.3 + g * 0.59 + b * 0.11),
+  a,
```

Comment:
seems alpha doesn't belong here
but I understand you want it the usage to be simple to use (less lines of code)
I suggest then returning a source value.
`[greyAvg, greyAvg, greyAvg, alpha]`

It is a bit weird like this, but not a big deal.

Reply:
yes maybe we can return the source format, is an internal util, it can do whatever it needs to do to make the code efficient

Reply:
we can make it from source to source.
Easier.